helix-rest/src/test/java/org/apache/helix/rest/metadatastore/TestShardingKeyTrieNode.java

```diff
@@ -0,0 +1,99 @@
+package org.apache.helix.rest.metadatastore;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.HashMap;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class TestShardingKeyTrieNode {
+
+  /**
+   * Testing the proper creation case, building a sharding key trie like so:
+   *     a
+   *    / \
+   *   b  e
+   *  / \
+   * c  d
+   */
+  @Test
+  public void testCreation() {
```

Comment:
Could we also add a "mock" write as if a parser was translating what's in ZK into a prefix tree? Some sort of tree traversal would be good to add here.

Reply:
Per offline discussion, the translation/parsing test will be provided in the parser class.